lib/fog/vsphere/models/compute/server.rb

```diff
@@ -319,6 +320,13 @@ def initialize_scsi_controller
             Fog::Compute::Vsphere::SCSIController.new(self.attributes[:scsi_controller])
           end
         end
+
+        def initialize_virtual_machine_summary
+          if attributes[:summary] && attributes[:summary].props
+            self.attributes[:summary] = Fog::Compute::Vsphere::VirtualMachineSummary.new(self.attributes[:summary].props)
```

Comment:
Yeah, so that feature was added because of `fog-xenserver`. Take a look at its models [here](https://github.com/fog/fog-xenserver/tree/master/lib/fog/compute/xen_server/models). None of the other providers did started using this feature yet but once they do lots of code won't be needed anymore.